drivers/crypto/crypto_stm32.c

```diff
@@ -408,8 +422,13 @@ static int crypto_stm32_session_free(const struct device *dev,
 		k_sem_give(&data->session_sem);
 		return -EIO;
 	}
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_cryp)
 	__HAL_RCC_CRYP_FORCE_RESET();
 	__HAL_RCC_CRYP_RELEASE_RESET();
+#elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32_aes)
+	__HAL_RCC_AES_FORCE_RESET();
+	__HAL_RCC_AES_RELEASE_RESET();
+#endif
```

Comment:
Nit: Provide abstraction macros here and for `xxx_TypeDef` to group the #idefery in a single place.

Reply:
I have now declared a few macros at the beginning of the file, please let me know if you had something different in mind.
Additionally I renamed `STM32_CRYP_KEYSIZE_192B_SUPPORT` to `STM32_CRYPTO_KEYSIZE_192B_SUPPORT`.